ndb/tests/unit/test_model.py

```diff
@@ -2722,6 +2656,55 @@ def test__put():
         with pytest.raises(NotImplementedError):
             entity._put()
 
+    @staticmethod
+    def test__lookup_model():
+        class ThisKind(model.Model):
+            pass
+
+        sentinel = object()
+        assert model.Model._lookup_model("ThisKind")
+        assert model.Model._lookup_model("NoKind", sentinel) is sentinel
```

Comment:
This test isn't actually deterministic. Some other code could load in a class called `NoKind` and then it would be in the kind map on `Model`. We unfortunately need something like `conftest.property_clean_cache()` to "reset" the internal state of `Model` to empty / clean.

Reply:
5f75adf